ciao-controller/main.go

```diff
@@ -283,3 +295,33 @@ func (c *controller) startCiaoService() error {
 
 	return nil
 }
+
+func getControllerFQDNFromCert(cert string) (string, error) {
+	bytesCert, err := ioutil.ReadFile(cert)
+	if err != nil {
+		return "", fmt.Errorf("couldn't read cert: %s", err)
+	}
+
+	blockCert, _ := pem.Decode(bytesCert)
+	if blockCert == nil {
+		return "", fmt.Errorf("couldn't decode cert")
+	}
+
+	parsedCert, err := x509.ParseCertificate(blockCert.Bytes)
+	if err != nil {
+		return "", fmt.Errorf("couldn't parse cert: %s", err)
+	}
+
+	role := ssntp.GetRoleFromOIDs(parsedCert.UnknownExtKeyUsage)
+	if !role.IsController() {
+		return "", fmt.Errorf("expected certificate role ssntp.Controller, got role %s", role.String())
```

Comment:
Your code is lgtm.

But i'm surprised you're taking it from the SSNTP cert. I guess that's fine but I would expected it to come from the HTTP cert that the server is actually served with. I think that should be present on clusterConfig.Configure.Controller. The only change should be for the path and to stop looking for the controller OID.

What do you think?